src/components/Main/Modal/SettingModal/ThemeTab/EditTheme.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div :class="$style.content">
+    <!--
+    <form-button
+      label="エクスポート"
+      @click="onImportClick"
+      :disabled="isImporterOpen"
+      color="secondary"
+      :class="$style.element"
+    />
+    -->
+    <form-button
+      label="インポート/エクスポート"
+      @click="onImportClick"
+      :disabled="isImporterOpen"
+      color="secondary"
+      :class="$style.element"
+    />
+    <!--
+    <form-button
+      label="選択中のテーマで上書き"
+      @click="onUpdateClick"
+      color="error"
+      :class="$style.element"
+    />
+    -->
+  </div>
+  <template v-if="isImporterOpen">
+    <textarea-autosize
+      @input-value="updateEditedTheme"
+      :value="editedTheme"
+      :class="$style.jsonField"
+    />
+    <div :class="$style.import">
+      <form-button
+        label="保存"
+        @click="onUpdateClick"
+        :disabled="isChanged"
+        color="primary"
+      />
+    </div>
+  </template>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, PropType, SetupContext } from 'vue'
+import FormButton from '@/components/UI/FormButton.vue'
+import store from '@/store'
+import { Theme } from '@/types/theme'
+import { dequal } from 'dequal'
+import { lightTheme } from '@/store/app/themeSettings/default'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isValidThemeJSON = (theme: any): theme is Theme => {
+  return (Object.keys(lightTheme) as Array<keyof Theme>).every(category =>
+    Object.keys(lightTheme[category]).every(
+      colorName =>
+        theme[category] &&
+        theme[category][colorName] &&
+        typeof theme[category][colorName] === 'string'
+    )
+  )
+}
+
+const failedUpdateTheme = (text?: string) => {
+  store.commit.ui.toast.addToast({
+    type: 'error',
+    text: 'テーマの更新に失敗しました' + (text === undefined ? '' : ': ' + text)
+  })
+}
+
+export default defineComponent({
+  name: 'EditTheme',
+  props: {
+    custom: {
+      type: Object as PropType<Theme>,
+      required: true
+    }
+  },
+  emits: {
+    'change-theme': (theme: Theme) => true
+  },
+  setup(props, context: SetupContext) {
```

Comment:
```suggestion
  setup(props) {
```
使われていないので受け取らなくてもいいです 👍

Reply:
これ嘘じゃんになってます 🙏